internal/jimm/jimm.go

```diff
@@ -524,438 +531,8 @@ func (j *JIMM) SSHManager() SSHManager {
-	RemoveCloud(context.Context, names.CloudTag) error
-
-	// RevokeApplicationOfferAccess revokes access to an application offer
-	// from a user.
-	RevokeApplicationOfferAccess(context.Context, string, names.UserTag, jujuparams.OfferAccessPermission) error
-
-	// RevokeCloudAccess revokes cloud access from a user.
-	RevokeCloudAccess(context.Context, names.CloudTag, names.UserTag, string) error
-
-	// RevokeCredential revokes a credential.
-	RevokeCredential(context.Context, names.CloudCredentialTag) error
-
-	// RevokeModelAccess revokes model access from a user.
-	RevokeModelAccess(context.Context, names.ModelTag, names.UserTag, jujuparams.UserAccessPermission) error
-
-	// SupportsCheckCredentialModels returns true if the
-	// CheckCredentialModels method can be used.
-	SupportsCheckCredentialModels() bool
-
-	// SupportsModelSummaryWatcher returns true if the connection supports
-	// a ModelSummaryWatcher.
-	SupportsModelSummaryWatcher() bool
-
-	// Status returns the status of the juju model.
-	Status(ctx context.Context, patterns []string) (*jujuparams.FullStatus, error)
-
-	// UpdateCloud updates a cloud definition.
-	UpdateCloud(context.Context, names.CloudTag, jujuparams.Cloud) error
-
-	// UpdateCredential updates a credential.
-	UpdateCredential(context.Context, jujuparams.TaggedCredential) ([]jujuparams.UpdateCredentialModelResult, error)
-
-	// ValidateModelUpgrade validates that a model can be upgraded.
-	ValidateModelUpgrade(context.Context, names.ModelTag, bool) error
-
-	// WatchAllModelSummaries creates a ModelSummaryWatcher.
-	WatchAllModelSummaries(context.Context) (string, error)
-
-	// ListFilesystems lists filesystems for desired machines.
-	// If no machines provided, a list of all filesystems is returned.
-	ListFilesystems(ctx context.Context, machines []string) ([]jujuparams.FilesystemDetailsListResult, error)
-
-	// ListVolumes lists volumes for desired machines.
-	// If no machines provided, a list of all volumes is returned.
-	ListVolumes(ctx context.Context, machines []string) ([]jujuparams.VolumeDetailsListResult, error)
-
-	// ListStorageDetails lists all storage.
-	ListStorageDetails(ctx context.Context) ([]jujuparams.StorageDetails, error)
-
-	// ListModels returns all UserModel's on the controller.
-	ListModels(ctx context.Context) ([]base.UserModel, error)
-}
-
-// forEachController runs a given function on multiple controllers
-// simultaneously. A connection is established to every controller in the
-// given list concurrently and then the given function is called with the
-// controller and API connection to use to perform the controller
-// operation. ForEachConnection waits until all operations have finished
-// before returning, any error returned will be the first error
-// encountered when connecting to the controller or returned from the given
-// function.
-func (j *JIMM) forEachController(ctx context.Context, controllers []dbmodel.Controller, f func(*dbmodel.Controller, API) error) error {
```

Comment:
The lines below are now inside of `internal/jimm/juju/jimm.go`. These methods are likely here because someone considered them "jimm-specific" logic associated with Juju controllers. But arguably many could move to different files like controllers.go or models.go, etc.